benchmarks/Csharp/Benchmarks/NBody.cs

```diff
@@ -0,0 +1,195 @@
+﻿namespace Harness.Benchmarks;
+
+public sealed class NBody : IBenchmark
+{
+    public bool Benchmark(int innerIterations)
+    {
+        var system = new NBodySystem();
+        for (var i = 0; i < innerIterations; i++)
+        {
+            system.Advance(0.01);
+        }
+
+        return VerifyResult(system.Energy(), innerIterations);
```

Comment:
This is not the same code as it's in any of the other versions.
Please stick closely to that code.